components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/builders/SignKeyDataHolder.java

```diff
@@ -101,12 +104,13 @@ public SignKeyDataHolder(String username) throws IdentityException {
 
             String signWithValue = IdentityUtil.getProperty(
                     SAMLSSOConstants.FileBasedSPConfig.USE_AUTHENTICATED_USER_DOMAIN_CRYPTO);
+            tenantDomain = spTenantDomain;
             if (signWithValue != null && "true".equalsIgnoreCase(signWithValue.trim())) {
                 tenantDomain = userTenantDomain;
+            }
+            tenantID = PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantId();
+            if (SAMLSSOUtil.getRealmService() != null) {
```

Comment:
`SAMLSSOUtil.getRealmService() == null` situation should throw a runtime exception as that mean this component is  not started correctly